ultraviolet/shared/src/test/scala/ultraviolet/acceptance/GLSLInterpolatorTests.scala

```diff
@@ -0,0 +1,21 @@
+package ultraviolet.acceptance
+
+import ultraviolet.syntax.*
+
+class GLSLInterpolatorTests extends munit.FunSuite {
+  test("interpolators") {
+    inline def fragment: Shader[Unit, vec4] =
+      Shader { _ =>
+        vec4(hex"ff00ff",0f)
+      }
+
+    val actual = fragment.toGLSL[WebGL2].toOutput.code
```

Comment:
This fails with the following compile-time error: `ultraviolet.datatypes.ShaderError$Unsupported: [ultraviolet] Shaders do not support infix operator: hex`.

I can see one of two locations where the error is thrown (in CreateShaderAST), but before I start investigating with DebugAST I was going to ask if you had any tips - I don't have much experience with macros in Scala 3, and I can't see why it thinks `hex` is an infix operator!

Reply:
Well done for getting this far!

Yes, so **The Big Macro** only accepts certain infix operators, and it does so with an explicit pattern match.

BEFORE ANYTHING ELSE! Try making your `hex` and `rgb` methods `inline`. The problem _might_ just go away...

Didn't work? Oh well... the process is: (I'm happy to help!)

1. Make the test (Done)
2. Identify the error (Done)
3. Find the error in the macro (Done)
4. Try a cheeky cargo-cult of what's there and see if it 'just works' (probably not but worth a try).
5. Pull out `DebugAST` and run it on your program
6. Copy paste the output into a worksheet you can format
7. Get a cup of tea or a coffee
8. Settle in and start reading, you're looking for bit with your `hex` in it.
9. Compare with the match statements and see why it isn't being caught.
10. Try and put in a match that works.

Good luck! Shout if you need backup!

Reply:
Thanks for the debugging tips! I'll give this a try and ping you if I need a hand!